pre_commit_hooks/file_contents_sorter.py

```diff
@@ -0,0 +1,57 @@
+"""
+A very simple pre-commit hook that, when passed one or more filenames
+as arguments, will sort the lines in those files.
+
+An example use case for this: you have a deploy-whitelist.txt file
+in a repo that contains a list of filenames that is used to specify
+files to be included in a docker container. This file has one filename
+per line. Various users are adding/removing lines from this file; using
+this hook on that file should reduce the instances of git merge
+conflicts and keep the file nicely ordered.
+"""
+from __future__ import print_function
+
+import argparse
+
+PASS = 0
+FAIL = 1
+
+
+def sort_file_contents(f):
+    before = [line for line in f]
+    after = sorted(before)
+
+    before_string = b''.join(before)
+    after_string = b''.join(after)
+
+    if before_string == after_string:
+        return PASS
+    else:
+        f.seek(0)
+        f.write(after_string)
+        f.truncate()
+        return FAIL
+
+
+def parse_commandline_input(argv):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('filenames', nargs='+', help='Files to sort')
+    args = parser.parse_args(argv)
+    return args
```

Comment:
I saw that most other hooks have the args parsed in their main function. Reasons I made it a separate function:
1) I found it made main() method more readable
2) unit testing

There's probably some I've-been-brainwashed-by-yelp_batch-which-does-this going on too.

Since the file is so short anyway, do you mind if I just leave it as a separate fn?